artichoke-backend/src/extn/core/string/trampoline.rs

```diff
@@ -586,10 +586,24 @@ pub fn byteindex(
     offset: Option<Value>,
 ) -> Result<Value, Error> {
     #[cfg(feature = "core-regexp")]
-    if let Ok(_pattern) = unsafe { Regexp::unbox_from_value(&mut substring, interp) } {
-        return Err(NotImplementedError::from("String#byteindex with Regexp pattern").into());
-    }
     let s = unsafe { super::String::unbox_from_value(&mut value, interp)? };
+    if let Ok(pattern) = unsafe { Regexp::unbox_from_value(&mut substring, interp) } {
+        let offset_number = match implicitly_convert_to_int(interp, offset.into()) {
+            Err(_) => None,
+            Ok(value) => Some(value),
+        };
+
+        let mut matches = pattern.match_(interp, Some(s.as_slice()), offset_number, None)?;
+        if matches.is_nil() {
+            return Ok(Value::nil());
+        }
+        let first_ocurrence = unsafe { MatchData::unbox_from_value(&mut matches, interp)? };
+        let ocurrence_index = first_ocurrence.begin(matchdata::Capture::GroupIndex(0)).unwrap();
```

Comment:
Please don't unwrap here. I know this is very much "can never panic" code, but the error types are compatible and it removes this "dangerous" call from the source.

```suggestion
        let ocurrence_index = first_ocurrence.begin(matchdata::Capture::GroupIndex(0))?;
```